Library/Homebrew/livecheck/strategy/crates.rb

```diff
@@ -0,0 +1,111 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      # The {Crates} strategy identifies versions of a crates.io package by
+      # checking the information from the `versions` API endpoint.
+      #
+      # Crate URLs have the following format:
+      #   `https://static.crates.io/crates/example/example-1.2.3.crate`
+      #
+      # The default regex identifies versions like `1.2.3`/`v1.2.3` from the
+      # version `num` field. This is a common version format but a different
+      # regex can be provided in a `livecheck` block to override the default
+      # if a package uses a different format (e.g. `1.2.3d`, `1.2.3-4`, etc.).
+      #
+      # @api public
+      class Crates
+        # The default regex used to identify versions when a regex isn't
+        # provided.
+        DEFAULT_REGEX = /^v?(\d+(?:\.\d+)+)$/i
+
+        # The default `strategy` block used to extract version information when
+        # a `strategy` block isn't provided.
+        DEFAULT_BLOCK = proc do |json, regex|
+          json["versions"]&.map do |version|
+            next if version["yanked"] == true
```

Comment:
```suggestion
            next if version["yanked"]
```

Reply:
I'm a little wary of using a truthy comparison (e.g., "false" would be seen as true) but it's fine if this will always be a boolean value. We can rely on that more if we're only working with crates.io but alternative registry implementations may not adhere strictly to the crates.io format (i.e., the Web API documentation doesn't contain specifics about version objects, `yanked`, etc.).

That said, we can always revisit this in the future if we run into issues from a truthy comparison, so I'll add this change on the next push.

Reply:
> e.g., "false" would be seen as true

Yes, but the same problem applies since `"true"` would be seen as “not yanked” currently.